src/commands/dev/generate/flag.ts

```diff
@@ -0,0 +1,262 @@
+/*
+ * Copyright (c) 2022, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as fs from 'fs/promises';
+import { SfCommand, Flags } from '@salesforce/sf-plugins-core';
+import { Messages } from '@salesforce/core';
+import { Nullable } from '@salesforce/ts-types';
+import { Duration } from '@salesforce/kit';
+
+import { Config, toConfiguredId, toStandardizedId } from '@oclif/core';
+import ModuleLoader from '@oclif/core/lib/module-loader';
+import * as fg from 'fast-glob';
+import { fileExists } from '../../../util';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.load('@salesforce/plugin-dev', 'dev.generate.flag', [
+  'description',
+  'errors.InvalidDir',
+  'examples',
+  'flags.dry-run.summary',
+  'errors.FlagExists',
+  'summary',
+]);
+
+type Answers = {
+  char: Nullable<string>;
+  type: keyof typeof Flags;
+  name: string;
+  required: boolean;
+  multiple: boolean;
+  durationUnit: Lowercase<keyof typeof Duration.Unit>;
+  durationDefaultValue: number;
+  salesforceIdLength: '15' | '18' | 'None';
+  salesforceIdStartsWith: string;
+  fileOrDirExists: boolean;
+  integerMin: number;
+  integerMax: number;
+};
+
+const toLowerKebabCase = (str: string): string =>
+  str
+    .replace(/([a-z])([A-Z])/g, '$1-$2')
+    .replace(/[\s_]+/g, '-')
+    .toLowerCase();
+
+export default class DevGenerateFlag extends SfCommand<void> {
+  public static enableJsonFlag = false;
+  public static summary = messages.getMessage('summary');
+  public static description = messages.getMessage('description');
+  public static examples = messages.getMessages('examples');
+
+  public static flags = {
+    'dry-run': Flags.boolean({
+      summary: messages.getMessage('flags.dry-run.summary'),
+      char: 'd',
+    }),
+  };
+
+  public async run(): Promise<void> {
+    const { flags } = await this.parse(DevGenerateFlag);
+
+    if (!fileExists('package.json')) throw messages.createError('errors.InvalidDir');
+
+    const ids = (await fg('src/commands/**/*.ts')).map((file) => {
+      const p = path.parse(file.replace(path.join('.', 'src', 'commands'), ''));
+      const topics = p.dir.split('/');
+      const command = p.name !== 'index' && p.name;
+      const id = [...topics, command].filter((f) => f).join(':');
+      return id === '' ? '.' : id;
+    });
+
+    const { command } = await this.prompt<{ command: string }>([
+      {
+        type: 'list',
+        name: 'command',
+        message: 'Which command is this for',
```

Comment:
This confused me for a minute because I thought that `bin/dev dev generate flag` was asking me for the command again (I had already passed it `flag`). I see now what it is asking. Although this will likely only be confusing when running this command on `plugin-dev` itself, it might help to change the message a bit. Maybe `Select command to add a flag to`